### ISSUE TITLE
 Real time Midi Clock messages are misinterpreted as Program/Bank Change messages. 

### ISSUE THREAD

**Issue by Falcosoft** [KaleidonKep99/OmniMIDI#139]
**Problem**
In FSMP 5.7 I have implemented sending Midi Clock messages that can be a handy feature e.g. in case of AutoDrum: 
https://youtu.be/bnTbNi6yPaE
But I have noticed that in case of enabled Midi clocks OmniMidi produces strange Program/Bank Changes randomly. I could also reproduce the problem with other DAWs/Sequencers in case of enabled Midi sync/clock messages. 

**Steps to reproduce**
You can test with FSMP and enabled 'Send Midi Start/Stop/Clock Messges' option. (details in the linked video)
http://falcosoft.hu/midiplayer_57_test.zip

**Expected behavior**
I think instead of processing real time messages they should be simply ignored by OmniMidi since Bassmidi does not support them at all anyway. 

**Screenshots/Videos**
https://youtu.be/z3C7wmCdgZw

**Environment**
 Not relevant.

**Additional context**
I think the problem is this code part in BufferSystem.h file:
```
BASS_MIDI_StreamEvents(
      OMStream, BASS_MIDI_EVENTS_RAW,
      &dwParam1, ((dwParam1 & 0xF0) >= 0xF8 && (dwParam1 & 0xF0) <= 0xFF) ? 1 : (((dwParam1 & 0xF0) == 0xC0 || (dwParam1 & 0xF0) == 0xD0) ? 2 : 3)
```
The tests for Program Change and Channel Pressure messages are OK, but the test for the real time messages never satisfies the condition. (dwParam1 & 0xF0) >= 0xF8 can never be true so real time messages will always be 3 bytes instead of the correct 1 byte and this problem can generate ghost messages that can be falsely interpreted as Program/Bank change messages with running status.

**Comment by KaleidonKep99** [comment]
I will add a check to ignore them in the next canary release.
Thanks for the report!

**Comment by Falcosoft** [comment]
Thanks,
I have also found some other issues but I do not know if I should create other items about them here.
Here's the Vogons link about the other problems:
https://www.vogons.org/viewtopic.php?f=24&t=63514&p=750391#p750391

**Comment by KaleidonKep99** [comment]
Done!
Tell me if everything's working properly now. 
I also added the buffer size warning in the ASIO device selection window.

DL: https://github.com/KeppySoftware/OmniMIDI/releases/tag/6.7.1.0

**Comment by Falcosoft** [comment]
Hi,
1. Real time Midi Clock message problem is fixed now, thanks.
2. I think this warning is enough and not much else can be done currently in OmniMidi alone. Maybe Ian at Un4seen could find out some workaround. 
3. While Channel 10 can be checked now, there is no effect of the selection. Channel 10 is never transposed. Also after closing and reopening the 'Change transpose value' dialog channel 10 is always unchecked.
I have uploaded a midi file (GM2_Ch10_Piano.mid) to Vogons that can be used for testing:
https://www.vogons.org/download/file.php?id=62046

**Comment by KaleidonKep99** [comment]
Weird, I tested it and it worked...
Maybe the MIDI itself uses long messages for them, which can not be handled by the transpose system.

**Comment by Falcosoft** [comment]
Hi,
No, it uses simple short messages. Actually it's very easy to tell what Midi messages a Midi file uses by opening the Event Viewer/Debugger panel in FSMP. I have made another video about the problem and you can also see in the video the exact Midi messages GM2_Ch10_Piano.mid referred above uses.
https://youtu.be/6zog8CtNZQg

**Comment by Falcosoft** [comment]
Maybe you tested with the 'Live' option enabled. If 'Live' button is pressed then the transpose is working until you press apply and the transpose dialog is closed. Then the applied value is forgotten by OmniMidi. As you can see in the video the next time you open the Transpose dialog Channel 10 is unchecked. So the problem can be that some variable related to Channel 10 is not persisted properly.

**Comment by KaleidonKep99** [comment]
Oh I see the issue now. It should be fixed in 6.7.1.1.

**Comment by KaleidonKep99** [comment]
Fixed.
